src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitedDate = date.split(fromFormat[3]);
+  const dateObject = {};
+  const dateResult = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = splitedDate[i];
+
+    if (fromFormat[i].includes('Y')) {
+      if (splitedDate[i].length === 2) {
+        if (splitedDate[i] < 30) {
+          dateObject.YYYY = '20' + splitedDate[i];
+        } else {
+          dateObject.YYYY = '19' + splitedDate[i];
+        }
+      } else {
+        dateObject.YYYY = splitedDate[i];
+      }
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i] === 'DD') {
+      dateResult.push(dateObject.DD);
+    } else if (toFormat[i] === 'MM') {
+      dateResult.push(dateObject.MM);
```

Comment:
You could use dynamic keys, something like this
```suggestion
      dateResult.push(dateObject[toFormat[i]]);
```
And you will have only one exception - `YY`